examples/tutorial/auto_parallel/demo_resnet50.py

```diff
@@ -0,0 +1,106 @@
+                                                num_steps=num_steps,
+                                                sample_points=sample_points,
+                                                free_memory=free_memory,
+                                                start_factor=start_factor)
+
+    # print summary
+    print("==============test summary==============")
+    for budget, peak, step in zip(budgets, peak_hist, step_hist):
+        print(f'memory budget: {budget:.3f} MB, peak memory: {peak:.3f} MB, step time: {step:.3f} MS')
+
+    # plot valid results
+    fig, axs = plt.subplots(1, 2, figsize=(16, 8))
+    valid_idx = step_hist.index(next(step for step in step_hist if step != float("inf")))
+
+    # plot peak memory vs. budget memory
+    axs[0].plot(budgets[valid_idx:], peak_hist[valid_idx:])
+    axs[0].plot([budgets[valid_idx], budgets[-1]], [budgets[valid_idx], budgets[-1]], linestyle='--')
+    axs[0].set_xlabel("Budget Memory (MB)")
+    axs[0].set_ylabel("Peak Memory (MB)")
+    axs[0].set_title("Peak Memory vs. Budget Memory")
+
+    # plot relative step time vs. budget memory
+    axs[1].plot(peak_hist[valid_idx:], [step_time / step_hist[-1] for step_time in step_hist[valid_idx:]])
+    axs[1].plot([peak_hist[valid_idx], peak_hist[-1]], [1.0, 1.0], linestyle='--')
+    axs[1].set_xlabel("Peak Memory (MB)")
+    axs[1].set_ylabel("Relative Step Time")
+    axs[1].set_title("Step Time vs. Peak Memory")
+    axs[1].set_ylim(0.8, 1.5)
+
+    # save plot
+    fig.savefig("resnet50_benchmark.png")
+
+
+def resnet50_benchmark(batch_size, num_steps, sample_points, free_memory, start_factor):
+    world_size = 1
+    run_func_module = partial(_resnet50_benchmark,
+                              world_size=world_size,
+                              port=free_port(),
+                              batch_size=batch_size,
+                              num_steps=num_steps,
+                              sample_points=sample_points,
+                              free_memory=free_memory,
+                              start_factor=start_factor)
+    mp.spawn(run_func_module, nprocs=world_size)
+
+
+if __name__ == "__main__":
+    parser = ArgumentParser("ResNet50 Auto Activation Benchmark")
+    parser.add_argument("--batch_size", type=int, default=128, help="batch size for benchmark, default 128")
+    parser.add_argument("--num_steps", type=int, default=5, help="number of test steps for benchmark, default 5")
+    parser.add_argument(
+        "--sample_points",
+        type=int,
+        default=15,
+        help=
+        "number of sample points for benchmark from start memory budget to maximum memory budget (free_memory), default 15"
+    )
+    parser.add_argument("--free_memory",
+                        type=int,
+                        default=11000 * 1024**2,
+                        help="maximum memory budget for benchmark, default maximum memory avialable of current device")
```

Comment:
Should be more straightforward, issit? Cuz our audience may not calculate 11000 * 1024**2 by themselves.

Reply:
Maybe in `MB` format

Reply:
I polish the help information here, I think it is clear enough

Reply:
Oops I forget the command should be typed in bash, so I change it into MB format